Submitted by:            Matt Burgess (matthew_at_linuxfromscratch_dot_org)
Date:                    2012-12-30
Initial Package Version: 8.20
Upstream Status:         Hunk 1 is from upstream gnulib.  Hunk 2 is LFS-specific.
Origin:                  Upstream/LFS
Description:             Fixes 2 test failures.

diff -Naur coreutils-8.20.orig/gnulib-tests/test-getlogin.c coreutils-8.20/gnulib-tests/test-getlogin.c
--- coreutils-8.20.orig/gnulib-tests/test-getlogin.c	2012-10-23 14:17:25.000000000 +0000
+++ coreutils-8.20/gnulib-tests/test-getlogin.c	2012-12-30 10:37:39.986578598 +0000
@@ -39,6 +39,13 @@
   buf = getlogin ();
   if (buf == NULL)
     {
+      if (errno == ENOENT)
+        {
+          /* This can happen on GNU/Linux.  */
+          fprintf (stderr, "Skipping test: no entry in utmp file.\n");
+          return 77;
+        }
+
       /* getlogin() fails when stdin is not connected to a tty.  */
       ASSERT (errno == ENOTTY
               || errno == EINVAL /* seen on Linux/SPARC */
diff -Naur coreutils-8.20.orig/tests/du/bind-mount-dir-cycle.sh coreutils-8.20/tests/du/bind-mount-dir-cycle.sh
--- coreutils-8.20.orig/tests/du/bind-mount-dir-cycle.sh	2012-10-23 14:14:12.000000000 +0000
+++ coreutils-8.20/tests/du/bind-mount-dir-cycle.sh	2012-12-30 10:39:11.918986774 +0000
@@ -17,7 +17,7 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 . "${srcdir=.}/tests/init.sh"; path_prepend_ ./src
-print_ver_ du
+skip_ 'unsupported /etc/mtab configuration, see bug#12730'
 require_root_
 
 cleanup_()
